usecases/objects/auto_schema.go

```diff
@@ -72,37 +76,47 @@ func (m *autoSchemaManager) autoSchema(ctx context.Context, principal *models.Pr
 		classes = append(classes, schema.UppercaseClassName(object.Class))
 	}
 
+	err := m.authorizer.Authorize(principal, authorization.READ, authorization.CollectionsMetadata(classes...)...)
+	if err != nil {
+		return 0, err
+	}
+
 	vclasses, err := m.schemaManager.GetCachedClass(ctx, principal, classes...)
 	if err != nil {
 		return 0, err
 	}
 
 	for _, object := range objects {
 		if object == nil {
-			return 0, fmt.Errorf(validation.ErrorMissingObject)
+			return 0, ErrInvalidUserInput{validation.ErrorMissingObject}
 		}
 
 		if len(object.Class) == 0 {
 			// stop performing auto schema
-			return 0, fmt.Errorf(validation.ErrorMissingClass)
+			return 0, ErrInvalidUserInput{validation.ErrorMissingClass}
 		}
 
 		object.Class = schema.UppercaseClassName(object.Class)
 
-		vclass := vclasses[schema.UppercaseClassName(object.Class)]
+		vclass := vclasses[object.Class]
```

Comment:
Is this because the class name was capitalised further up in the call stack?

Reply:
directly the line above

Reply:
hahahaha, my brain